### ISSUE TITLE
When running loader with a single file it doesn't work

### ISSUE THREAD

**Issue by edeustace** [edeustace/assets-loader#16]
Note: Lifted from #13 

For example when I use this (I have only one css file with name style.css):
@views.Helper.loader.css("some_name")("stylesheets/style.css").body

I get incorrect path: BASE_URL/assets/stylesheets/style.css/some_name-0123456789.min.gz.css

**Comment by edeustace** [comment]
@ jkypet - I can't reproduce with https://github.com/jkypet/assets-loader-test

I tried :

```
play clean update run
play stage
play dist
play start
```

On OS X.

Could you provide some more information on how its run?

**Comment by twiho** [comment]
I've just tried,

```
play clean
play start
```

and I see

```
<link rel="stylesheet" type="text/css" href="/assets/stylesheets/styles.css/stylesheets/test-1848101319.min.gz.css"/>
```

in the source which isn't the correct link - thus this prevents using relative paths.

**Comment by edeustace** [comment]
odd - will try it later.

**Comment by edeustace** [comment]
Found the issue - should be fixed in 11.7-SNAPSHOT

**Comment by twiho** [comment]
Thank you! Please, let us know when you make it available, we'll try it!

**Comment by edeustace** [comment]
SNAPSHOT is published - let me know if it works and I'll release 11.7

**Comment by twiho** [comment]
I can't seem to be able to download it. What am I missing?

```
[info] Resolving com.ee#assets-loader_2.10;0.11.7-SNAPSHOT ...
[warn]  module not found: com.ee#assets-loader_2.10;0.11.7-SNAPSHOT
[warn] ==== local: tried
[warn]   /Applications/play-2.2.1/repository/local/com.ee/assets-loader_2.10/0.11.7-SNAPSHOT/ivys/ivy.xml
[warn] ==== Maven2 Local: tried
[warn]   file:/Users/twiho/.m2/repository/com/ee/assets-loader_2.10/0.11.7-SNAPSHOT/assets-loader_2.10-0.11.7-SNAPSHOT.pom
[warn] ==== public: tried
[warn]   http://repo1.maven.org/maven2/com/ee/assets-loader_2.10/0.11.7-SNAPSHOT/assets-loader_2.10-0.11.7-SNAPSHOT.pom
[warn] ==== Typesafe Releases Repository: tried
[warn]   http://repo.typesafe.com/typesafe/releases/com/ee/assets-loader_2.10/0.11.7-SNAPSHOT/assets-loader_2.10-0.11.7-SNAPSHOT.pom
[warn] ==== My GitHub Play Repository: tried
[warn]   http://branaway.github.com/releases/com.ee/assets-loader_2.10/0.11.7-SNAPSHOT/ivys/ivy.xml
[warn] ==== ed eustace: tried
[warn]   http://edeustace.com/repository/releases/com/ee/assets-loader_2.10/0.11.7-SNAPSHOT/assets-loader_2.10-0.11.7-SNAPSHOT.pom
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  ::          UNRESOLVED DEPENDENCIES         ::
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
[warn]  :: com.ee#assets-loader_2.10;0.11.7-SNAPSHOT: not found
[warn]  ::::::::::::::::::::::::::::::::::::::::::::::
sbt.ResolveException: unresolved dependency: com.ee#assets-loader_2.10;0.11.7-SNAPSHOT: not found
```

**Comment by edeustace** [comment]
Try again - should be there:

eg:
http://edeustace.com/repository/snapshots/com/ee/assets-loader_2.10/0.11.7-SNAPSHOT/assets-loader_2.10-0.11.7-SNAPSHOT.pom
is working

**Comment by twiho** [comment]
I'm sorry, misconfiguration - my bad.
Yay, can confirm it working! Thank you kind sir!

**Comment by edeustace** [comment]
You're welcome :)

**Comment by phedoreanu** [comment]
Hey @edeustace,

I get the same error as @larousso, in https://github.com/edeustace/assets-loader/issues/10#issue-24440120, when deploying the Play app in a Docker container:

```
play.api.Application$$anon$1: Execution exception[[MalformedInputException: Input length = 1]]
    at play.api.Application$class.handleError(Application.scala:289) ~[play.play_2.10-2.1.3.jar:2.1.3]
    at play.api.DefaultApplication.handleError(Application.scala:383) [play.play_2.10-2.1.3.jar:2.1.3]
    at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$18$$anonfun$apply$18.apply(PlayDefaultUpstreamHandler.scala:314) [play.play_2.10-2.1.3.jar:2.1.3]
    at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$18$$anonfun$apply$18.apply(PlayDefaultUpstreamHandler.scala:312) [play.play_2.10-2.1.3.jar:2.1.3]
    at play.api.libs.concurrent.PlayPromise$$anonfun$extend1$1.apply(Promise.scala:113) [play.play_2.10-2.1.3.jar:2.1.3]
    at play.api.libs.concurrent.PlayPromise$$anonfun$extend1$1.apply(Promise.scala:113) [play.play_2.10-2.1.3.jar:2.1.3]
java.nio.charset.MalformedInputException: Input length = 1
    at java.nio.charset.CoderResult.throwException(CoderResult.java:277) ~[na:1.7.0_65]
    at sun.nio.cs.StreamDecoder.implRead(StreamDecoder.java:338) ~[na:1.7.0_65]
    at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:177) ~[na:1.7.0_65]
    at java.io.InputStreamReader.read(InputStreamReader.java:184) ~[na:1.7.0_65]
    at java.io.BufferedReader.fill(BufferedReader.java:154) ~[na:1.7.0_65]
    at java.io.BufferedReader.readLine(BufferedReader.java:317) ~[na:1.7.0_65]
```

Works fine if deployed on a normal Linux machine. Do I just need to upgrade to the latest Play version?
Or am I missing some package to install?

Thank you,
Adrian

**Comment by edeustace** [comment]
@phedoreanu - I suspect this is a file encoding issue. assets-loader isn't available for play 2.3 or higher at the moment I'm afraid. I don't have much free time to look at the issue, but if you could prepare some steps to reproduce, I'll try and have a quick look. Either the library isn't loading explicitly as utf-8 or the file to load isn't encoded as utf-8 is my guess.

**Comment by phedoreanu** [comment]
Never mind, I resolved the issue adding `-Dfile.encoding=UTF-8` to `./start`. Thanks for the tip! ;)

Edit: Output from `locale` inside the Docker container:

```
root@1429ff954639:/opt# locale 
LANG=
LANGUAGE=
LC_CTYPE="POSIX"
LC_NUMERIC="POSIX"
LC_TIME="POSIX"
LC_COLLATE="POSIX"
LC_MONETARY="POSIX"
LC_MESSAGES="POSIX"
LC_PAPER="POSIX"
LC_NAME="POSIX"
LC_ADDRESS="POSIX"
LC_TELEPHONE="POSIX"
LC_MEASUREMENT="POSIX"
LC_IDENTIFICATION="POSIX"
LC_ALL=
```
